aten/src/ATen/native/cpu/PowKernel.cpp

```diff
@@ -35,66 +37,115 @@ void pow_tensor_tensor_kernel(TensorIterator& iter) {
-      using Vec = Vec256<scalar_t>;
-      if (exp == 0.5) {
-        cpu_kernel_vec(iter,
-          [](scalar_t base) -> scalar_t {
-            return std::sqrt(base);
-          },
-          [](Vec base) -> Vec { return base.sqrt(); }
-        );
-      } else if (exp == 2) {
+    if ((exp == 0.5) && (dtype != ScalarType::Half) && (dtype != ScalarType::BFloat16)) {
+      AT_DISPATCH_FLOATING_TYPES(dtype, "pow", [&]() {
+        using Vec = Vec256<scalar_t>;
+          cpu_kernel_vec(iter,
+            [](scalar_t base) -> scalar_t {
+              return std::sqrt(base);
+            },
+            [](Vec base) -> Vec { return base.sqrt(); }
+          );
+      });
+    } else if ((exp == 2) && (dtype != ScalarType::Half) && (dtype != ScalarType::BFloat16)) {
+      AT_DISPATCH_FLOATING_TYPES(dtype, "pow", [&]() {
+        using Vec = Vec256<scalar_t>;
         cpu_kernel_vec(iter,
           [](scalar_t base) -> scalar_t {
             return base * base;
           },
           [](Vec base) -> Vec { return base * base; }
         );
-      } else if (exp == 3) {
+      });
+    } else if ((exp == 3) && (dtype != ScalarType::Half) && (dtype != ScalarType::BFloat16)) {
+      AT_DISPATCH_FLOATING_TYPES(dtype, "pow", [&]() {
+        using Vec = Vec256<scalar_t>;
         cpu_kernel_vec(iter,
           [](scalar_t base) -> scalar_t {
             return base * base * base;
           },
           [](Vec base) -> Vec { return base * base * base; }
         );
-      } else if (exp == -0.5) {
+      });
+    } else if ((exp == -0.5) && (dtype != ScalarType::Half) && (dtype != ScalarType::BFloat16)) {
+      AT_DISPATCH_FLOATING_TYPES(dtype, "pow", [&]() {
+        using Vec = Vec256<scalar_t>;
         cpu_kernel_vec(iter,
           [](scalar_t base) __ubsan_ignore_float_divide_by_zero__ -> scalar_t {
             return 1.0 / std::sqrt(base);
           },
           [](Vec base) -> Vec { return base.rsqrt(); }
         );
-      } else if (exp == -1) {
+      });
+    } else if ((exp == -1) && (dtype != ScalarType::Half) && (dtype != ScalarType::BFloat16)) {
+      AT_DISPATCH_FLOATING_TYPES(dtype, "pow", [&]() {
+        using Vec = Vec256<scalar_t>;
         cpu_kernel_vec(iter,
           [](scalar_t base) -> scalar_t {
             return 1.0 / base;
           },
           [](Vec base) -> Vec { return base.reciprocal(); }
         );
-      } else if (exp == -2) {
+      });
+    } else if ((exp == -2) && (dtype != ScalarType::Half) && (dtype != ScalarType::BFloat16)) {
+      AT_DISPATCH_FLOATING_TYPES(dtype, "pow", [&]() {
+        using Vec = Vec256<scalar_t>;
         cpu_kernel_vec(iter,
           [](scalar_t base) -> scalar_t {
             return 1.0 / (base * base);
           },
           [](Vec base) -> Vec { return (base * base).reciprocal(); }
         );
+      });
+    } else {
+      // Casting exp to double isn't possible with Half & BFloat16 common_dtype(),
```

Comment:
Would you elaborate on this limitation? Why can't exp be cast to double?

Reply:
I initially thought you meant I should elaborate in the comment.
But if you were asking me to elaborate here, it's because we'd run into compilation errors such as,

```
error: no matching function for call to ‘at::vec256::{anonymous}::Vec256<c10::BFloat16>::pow(const double&)’
           [=](Vec base) -> Vec { return base.pow(exp); }
```

```
note:   no known conversion for argument 1 from ‘const double’ to ‘const at::vec256::{anonymous}::Vec256<c10::BFloat16>&’

```

I'll mention it concisely in the comment as well.